### ISSUE TITLE
recipe for Thriftgens failed

### ISSUE THREAD

**Issue by nishtala** [uber/peloton#11]
I'm trying to setup Peloton on a Linux machine.
go version go1.10.4 linux/amd64

The recipe for thirftgens fails here
....
go get ./vendor/go.uber.org/thriftrw
vendor/go.uber.org/thriftrw/main.go:34:2: use of internal package not allowed
vendor/go.uber.org/thriftrw/main.go:35:2: use of internal package not allowed
Makefile:117: recipe for target 'thriftgens' failed
make: *** [thriftgens] Error 1

what am I missing?

**Comment by amittbose** [comment]
Can you try a later version of Go, e.g. 1.12.4?

**Comment by nishtala** [comment]
with Go version 1.12.5, this is the following error
```
go get ./vendor/go.uber.org/thriftrw
vendor/go.uber.org/thriftrw/main.go:34:2: use of internal package go.uber.org/thriftrw/internal/plugin not allowed
vendor/go.uber.org/thriftrw/main.go:35:2: use of internal package go.uber.org/thriftrw/internal/plugin/builtin/pluginapigen not allowed
Makefile:117: recipe for target 'thriftgens' failed

```

**Comment by ublackcat** [comment]
@nishtala can you make sure the peloton repo is cloned inside the gopath, e.g.

```
export GOPATH="$HOME/gopath"
export PATH="$PATH:$GOPATH/bin"
mkdir -p "$GOPATH/src/github.com/uber"
git clone https://github.com/uber/peloton.git "$GOPATH/src/github.com/uber/peloton"
```

**Comment by nishtala** [comment]
Yes, it compiles with protoc 3.8.0 and go 1.12.5. But setting it up in the Go path is inconvenient. Two more things (1) is there a publication for peloton? (2) The slack channel is down.

**Comment by talaniz** [comment]
@nishtala could you please provide more information about the use case related to this comment? 
> But setting it up in the Go path is inconvenient.

What kind of publications did you have in mind? Would a newsletter be of interest? Also, what slack channel are you accessing? It looks like the slack channel is up but I'd like to verify which one you're using.

**Comment by nishtala** [comment]
A publication for this talk: https://www.youtube.com/watch?v=USgbj87Ztlk
A newsletter would be interesting.
This slack channel: https://github.com/uber/peloton#contact

**Comment by talaniz** [comment]
@nishtala I will work on getting a newsletter or communication going. I have tested the slack link and see that it's working on my end now. Can you confirm if it's working for you?

Also, would you be able to help me understand the use case for setting up outside of the gopath? I'd like to document ways that users would need something like this outside of the recommended setup.

**Comment by nishtala** [comment]
Re: Slack. "This invite link is no longer active. Please check with the person who shared it with you to see if there's a new link available."

Re: Go. I don't crisply recall what is the exact changes I did. I can dig into it in two weeks time and get back to you -  if that works.
